format/src/matbin/mod.rs

```diff
@@ -0,0 +1,309 @@
+use std::{borrow::Cow, io};
+
+use bytemuck::PodCastError;
+use byteorder::LE;
+use thiserror::Error;
+use widestring::{U16Str, U16String};
+use zerocopy::{FromBytes, FromZeroes, Ref, F32, U32, U64};
+
+use crate::io_ext::zerocopy::Padding;
+
+#[derive(Debug, Error)]
+pub enum MatbinError {
+    #[error("Could not copy bytes {0}")]
+    IO(#[from] io::Error),
+
+    #[error("Could not read string")]
+    String(#[from] ReadUtf16StringError),
+
+    #[error("Got unknown parameter type {0}")]
+    UnknownParameterType(u32),
+
+    #[error("Could not create reference to value")]
+    ValueZerocopy,
+}
+
+#[derive(FromZeroes, FromBytes)]
```

Comment:
Remove `FromBytes` and `FromZeroes` on this impl, it's not a type we can grab from a slice of bytes. Also no need for it to be `repr(C)`.

Reply:
574b93d216ec8d05173f6aeb66908c934b48df2f removes it